src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -392,52 +462,58 @@ private static CodeActionKind GetCodeActionKindFromSuggestedActionCategoryName(s
                 _ => throw ExceptionUtilities.UnexpectedValue(priority)
             };
 
-        public static CodeAction? GetCodeActionToResolve(string distinctTitle, ImmutableArray<CodeAction> codeActions)
+        public static CodeAction? GetCodeActionToResolve(string[] codeActionPath, ImmutableArray<CodeAction> codeActions, bool isFixAllAction)
         {
-            // Searching for the matching code action. We compare against the unique identifier
-            // (e.g. "Suppress or Configure issues|Configure IDExxxx|Warning") instead of the
-            // code action's title (e.g. "Warning") since there's a chance that multiple code
-            // actions may have the same title (e.g. there could be multiple code actions with
-            // the title "Warning" that appear in the code action menu if there are multiple
-            // diagnostics on the same line).
-            foreach (var codeAction in codeActions)
-            {
-                var action = CheckForMatchingAction(codeAction, distinctTitle);
-                if (action != null)
-                    return action;
-            }
-
-            return null;
+            return FindCodeAction(codeActions, codeActionPath, isFixAllAction, 0);
         }
 
-        private static CodeAction? CheckForMatchingAction(CodeAction codeAction, string goalTitle, string currentTitle = "")
+        private static CodeAction? FindCodeAction(ImmutableArray<CodeAction> codeActions, string[] titlePath, bool isFixAllAction, int index)
         {
-            // If the unique identifier of the current code action matches the unique identifier of the code action
-            // we're looking for, return the code action. If not, check to see if one of the current code action's
-            // nested actions may be a match.
+            if (index < 0 || index >= titlePath.Length)
+            {
+                return null;
+            }
 
-            if (!string.IsNullOrEmpty(currentTitle))
+            CodeAction? matchingAction = null;
+            foreach (var action in codeActions)
             {
-                // Adding a delimiter for nested code actions, e.g. 'Suppress or Configure issues.Suppress IDEXXXX|in Source'
-                currentTitle += '|';
+                if (action.Title == titlePath[index])
+                {
+                    matchingAction = action;
+                    if (index == titlePath.Length - 1) 
```

Comment:
Not quite getting the length check.  Naively it seems like if we found a matching title then we don't need to do anything else right?  Why specifically check here if we're at the end - in the code below outside the loop it already checks if we're at the end before recursing.